imjoy_elfinder/elfinder.py

```diff
@@ -759,47 +767,135 @@ def __upload(self) -> None:
             total = 0
             up_size = 0
             max_size = self._options["upload_max_size"]
-            for name, data in up_files.items():
-                if name:
-                    name = self._check_utf8(name)
-                    total += 1
-                    name = os.path.basename(name)
-                    if not _check_name(name):
-                        self._set_error_data(name, "Invalid name: " + name)
+            chunk = self._request.get(API_CHUNK)
+            if chunk:
```

Comment:
I would try to split this method up and see if that can make it easier to read.

Reply:
Yeah, the implementation and if statements are horrible at the moment, I try to make it work.

Still not perfect at the moment, when uploading large files, I cannot make the upload stop even if I set "warning". The current version silently shutdown if any error occurred, not sure how to solve yet.

We need some refactor with the upload function, could you please take a look see how to improve it? @MartinHjelmare

Reply:
I can check later in the week.

Reply:
I will merge this for now, let's refactor it in another PR.